src/components/GroupList/GroupItem.js

```diff
@@ -0,0 +1,61 @@
+/* @flow */
+
+import React from 'react'
+import Divider from 'material-ui/Divider'
+import Avatar from 'material-ui/Avatar'
+import FileFolder from 'material-ui/svg-icons/file/folder'
+import { ListItem } from 'material-ui/List'
+import type { GroupType } from 'ducks/projects'
+
+export type Props = {
+  group: GroupType,
+  inset: number,
+  clickHandler: Function,
+}
+
+
+const GroupItem = (props : Props) => {
+  const {
+    group,
+    clickHandler,
+    inset,
+  } = props
+
+  const {
+    name,
+    description,
+    id,
+  } = group
+
+  function touchTapEventHandler() {
+    if (clickHandler) {
```

Comment:
I wouldn't be afraid that much react classes, stateless functional components wrapped in a class internally :) They go through the same lifecycle as react classes. So from the performance perspective, there is no advantage of not using react classes. Probably there is more code style advantage of writing less noise code. But we should keep a wise balance here.

There could be possibly many items. And each of those would create a new function on rendering every time.  So for this case, I would go either with requiring `clickHandler` and `onTouchTap={props.clickHandler}`(which I am not a fan of) or we are doing the component where we bind `touchTapEventHandler`. I would recommend here rather class.